File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int n=s.length(),a=0,b=0,c=0;
        int ans=0;
        for(int i=0,j=0;j<n;j++){
            if(s[j]=='a') a++;
            else if(s[j]=='b') b++;
            else c++;
            while(a && b && c){
                ans+= n-j;
                if(s[i]=='a') a--;
                else if(s[i]=='b') b--;
                else c--;
                i++;
            }
        }
        return ans;
    }
};

File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        n=len(s);a=0;b=0;c=0;
        ans=0;i=0;
        for j in range(n):
            if(s[j]=='a'):
                a+=1
            elif(s[j]=='b'):
                b+=1
            else:
                c+=1
            while(a and b and c):
                ans+= n-j
                if(s[i]=='a'):
                    a-=1
                elif(s[i]=='b'):
                    b-=1
                else:
                    c-=1
                i+=1
        return ans